catalog/spatial/registry/registry-api-impl/src/test/java/org/codice/ddf/registry/api/impl/RegistryStoreCleanupHandlerTest.java

```diff
@@ -0,0 +1,238 @@
+        handleEvent(null,
+                ref,
+                ServiceEvent.UNREGISTERING,
+                "registryId",
+                Collections.singletonList(mcard));
+        verify(federationAdmin,
+                times(1)).deleteRegistryEntriesByMetacardIds(Collections.singletonList(mcard.getId()));
+    }
+
+    @Test
+    public void testUnregisterEvent() throws Exception {
+        ServiceReference ref = getServiceReference("servicePid");
+        handleEvent(ref,
+                ref,
+                ServiceEvent.UNREGISTERING,
+                "registryId",
+                Collections.singletonList(mcard));
+        verify(federationAdmin).deleteRegistryEntriesByMetacardIds(Collections.singletonList(mcard.getId()));
+    }
+
+    @Test
+    public void testUnregisterEventCleanupDisabled() throws Exception {
+        cleanupHandler.setCleanupRelatedMetacards(false);
+        ServiceReference ref = getServiceReference("servicePid");
+        handleEvent(ref,
+                ref,
+                ServiceEvent.UNREGISTERING,
+                "registryId",
+                Collections.singletonList(mcard));
+        verify(federationAdmin,
+                never()).deleteRegistryEntriesByMetacardIds(Collections.singletonList(mcard.getId()));
+    }
+
+    @Test
+    public void testUnregisterEventWrongServiceEvent() throws Exception {
+        ServiceReference ref = getServiceReference("servicePid");
+        handleEvent(ref,
+                ref,
+                ServiceEvent.REGISTERED,
+                "registryId",
+                Collections.singletonList(mcard));
+        verify(federationAdmin,
+                never()).deleteRegistryEntriesByMetacardIds(Collections.singletonList(mcard.getId()));
+    }
+
+    @Test
+    public void testUnregisterEventUnrelatedService() throws Exception {
+        ServiceReference ref1 = getServiceReference("servicePid");
+        ServiceReference ref2 = getServiceReference("differentServicePid");
+        handleEvent(ref1,
+                ref2,
+                ServiceEvent.UNREGISTERING,
+                "registryId",
+                Collections.singletonList(mcard));
+        verify(federationAdmin,
+                never()).deleteRegistryEntriesByMetacardIds(Collections.singletonList(mcard.getId()));
+    }
+
+    @Test
+    public void testUnregisterEventNoRelatedMetacards() throws Exception {
+        ServiceReference ref = getServiceReference("servicePid");
+        handleEvent(ref, ref, ServiceEvent.UNREGISTERING, "registryId", Collections.emptyList());
+        verify(federationAdmin,
+                never()).deleteRegistryEntriesByMetacardIds(Collections.singletonList(mcard.getId()));
+    }
+
+    @Test
+    public void testDestroy() throws Exception {
+        when(executorService.awaitTermination(anyLong(), any(TimeUnit.class))).thenReturn(true);
+        cleanupHandler.destroy();
+        verify(executorService, times(1)).awaitTermination(anyLong(), any(TimeUnit.class));
```

Comment:
Should we have at least one test that uses real numbers instead of `anyLong()` and `any(TimeUnit.class)` to make sure no one ever changes one of those by mistake and uses values and/or units that are way too large or small?